### PR TITLE
add sample checkout flow

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

File: backend/checkout.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	if r.Method == "GET" {
+		clientId := r.URL.Query().Get("clientId")
+		writeShoppingCart(w, r, clientId)
+	}
+}
+
+func writeShoppingCart(w http.ResponseWriter, r *http.Request, clientId string) {
+	w.Header().Set("Content-Type", "application/json")
+	discount := discounts[clientId]
+	if discount == 0 {
```
Comment (kul3r4) [inline backend/checkout.go]: why is this happening?

File: backend/checkout.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		discount = 1
+	}
+	total := 9.94 * discount
+	cart := map[string]interface{}{
```
Comment (kul3r4) [inline backend/checkout.go]: Can you move this into a method like "createCart"

File: backend/checkout.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+				"quantity": "1",
+			},
+			map[string]interface{}{
+				"name":     "Item 3",
+				"price":    "0.99",
+				"quantity": "3",
+			},
+		},
+	}
+	if discount != 1 {
```
Comment (kul3r4) [inline backend/checkout.go]: what does it mean being a discount of value 1?

File: src/50_Samples_%26_Templates/Checkout_Flow.html [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+  <!-- -->
+  <!doctype html>
+  <html ⚡>
+    <head>
+      <meta charset="utf-8">
+      <link rel="canonical" href="<%host%>/../samples_templates/checkout_flow/">
+      <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+      <script async src="https://cdn.ampproject.org/v0.js"></script>
+
+      <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.1.js"></script>
```
Comment (kul3r4) [inline src/50_Samples_%26_Templates/Checkout_Flow.html]: In previous samples we added a line explaining each import, do you want to follow this convention?

File: src/50_Samples_%26_Templates/Checkout_Flow.html [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+      <div class="my2" [class]="checkoutSuccess ? 'hide' : 'my2'" >
+        <button amp-access="NOT loggedIn" on="tap:amp-access.login-sign-in" class="ampstart-btn ampstart-btn-secondary ml1 caps">Login</button> 
+        <button amp-access="loggedIn" amp-access-hide tabindex="0" on="tap:amp-access.login-sign-out" class="ampstart-btn ml1 caps" amp-access-hide>Logout</button>
+        <span class="inline-block ml1">...or continue as guest</span>
+      </div>
+
+      <!-- ## Review Order (Shopping Cart) -->
+      <!-- The review order section displays the current shopping cart content and the total price. We pull in the shopping cart content from a JSON endooint and render it inside an `amp-list`. The user is identified based on the AMP CLIENT_ID which is passed as
```
Comment (kul3r4) [inline src/50_Samples_%26_Templates/Checkout_Flow.html]: typo: endooint -> endpoint

File: src/50_Samples_%26_Templates/Checkout_Flow.html [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+        a request parameter. Additionally, We bind the amp-list's `src` attribute to an implicit state object `[src]="shoppingCart.src"`, which we use to refresh the content using `amp-bind`.
```
Comment (kul3r4) [inline src/50_Samples_%26_Templates/Checkout_Flow.html]: We -> we

File: src/50_Samples_%26_Templates/Checkout_Flow.html [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+              <input name="ship-state" id="frmStateS" placeholder="NY" autocomplete="shipping region">
+
+              <label for="frmZipS">Zip</label>
+              <input name="ship-zip" id="frmZipS" placeholder="10011" autocomplete="shipping postal-code">
+
+              <label for="frmCountryS">Country</label>
+              <input name="ship-country" id="frmCountryS" placeholder="USA" autocomplete="shipping country">
+            </section>
+
+            <!-- ## Payment Details -->
```
Comment (kul3r4) [inline src/50_Samples_%26_Templates/Checkout_Flow.html]: similiar -> similar

File: src/50_Samples_%26_Templates/Checkout_Flow.html (hunk 5 above)
Comment (sebastianbenz) [reply src/50_Samples_%26_Templates/Checkout_Flow.html]: done

File: src/50_Samples_%26_Templates/Checkout_Flow.html (hunk 6 above)
Comment (sebastianbenz) [reply src/50_Samples_%26_Templates/Checkout_Flow.html]: done

File: src/50_Samples_%26_Templates/Checkout_Flow.html (hunk 7 above)
Comment (sebastianbenz) [reply src/50_Samples_%26_Templates/Checkout_Flow.html]: one of the words I never get right...done

File: backend/checkout.go (hunk 2 above)
Comment (sebastianbenz) [reply backend/checkout.go]: done

File: backend/checkout.go (hunk 1 above)
Comment (sebastianbenz) [reply backend/checkout.go]: refactored the whole section, should be hopefully clearer now

File: backend/checkout.go (hunk 3 above)
Comment (sebastianbenz) [reply backend/checkout.go]: removed

File: src/50_Samples_%26_Templates/Checkout_Flow.html (hunk 4 above)
Comment (sebastianbenz) [reply src/50_Samples_%26_Templates/Checkout_Flow.html]: yes, you're right

**Comment by sebastianbenz** [comment]
Addressed the comments. PTAL

**Comment by kul3r4** [comment]
Approved! take a look at the failing CI before merging.

**Comment by sebastianbenz** [comment]
Thanks for the review @kul3r4 !

**Pull request merged**